src/main/scala/lambdaone/kratia/registry/Registry.scala

```diff
@@ -0,0 +1,41 @@
+package lambdaone.kratia.registry
+
+import cats.Monad
+import lambdaone.toolbox.{EventStore, Store}
+import cats.implicits._
+
+trait Registry[F[_], A] {
+
+  def isMember(community: Community[A], member: Member[A]): F[Boolean]
+
+  def load(community: Community[A], member: Member[A]): F[Option[A]]
+
+  def loadAll(community: Community[A]): F[List[A]]
+
+  def register(community: Community[A], member: Member[A], data: A): F[()]
+}
+
+object Registry {
+
+  implicit def registryWithEventStore[F[_], A](implicit store: EventStore[F, RegistryEvent[A]], F: Monad[F]): Registry[F, A] =
```

Comment:
Here we are writing the Registry algebra in terms of an EventStore algebra. Note that we are adding a restriction to the EventStore algebra, we are saying: "We are able to provide a Registry algebra implementation, given an EventAlgebra which is implemented through the same F and through a RegistryEvent of A"